platform/ui/src/components/HeaderPatientInfo/HeaderPatientInfo.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from '@ohif/ui';
+import { utils } from '@ohif/core';
+import { useAppConfig } from '@state';
+
+const { formatDate, formatPN } = utils;
+
+function usePatientInfo(servicesManager) {
+  const [patientInfo, setPatientInfo] = useState({
+    PatientName: '',
+    PatientID: '',
+    PatientSex: '',
+    PatientAge: '',
+    PatientDOB: '',
+  });
+  const [isMixedPatients, setIsMixedPatients] = useState(false);
+
+  const { displaySetService, viewportGridService, cornerstoneViewportService } =
+    servicesManager.services;
+
+  const { activeViewportId, viewports } = viewportGridService.getState();
+
+  const checkMixedPatients = activeViewportPatientID => {
+    const { viewports } = viewportGridService.getState();
+    let isMixedPatients = false;
+    viewports.forEach(viewport => {
+      const displaySetInstanceUID = viewport.displaySetInstanceUIDs[0];
+      if (!displaySetInstanceUID) {
+        return;
+      }
+      const displaySet = displaySetService.getDisplaySetByUID(displaySetInstanceUID);
+      const instance = displaySet?.instances?.[0] || displaySet?.instance;
+      if (!instance) {
+        return;
+      }
+
+      if (instance.PatientID !== activeViewportPatientID) {
+        isMixedPatients = true;
+      }
+    });
+    setIsMixedPatients(isMixedPatients);
+  };
```

Comment:
I guess you should just list all DSs and look if there is inconsistency, not just viewports. To be careful here